contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -323,28 +325,60 @@ where
 		process_ack_packet(&mut ctx, packet, &packet_data, &ack)
 			.map_err(|e| Ics04Error::implementation_specific(e.to_string()))?;
 		match ack.into_result() {
-			Ok(_) => Pallet::<T>::deposit_event(Event::<T>::TokenTransferCompleted {
-				from: packet_data.sender,
-				to: packet_data.receiver,
-				ibc_denom: packet_data.token.denom.to_string().as_bytes().to_vec(),
-				local_asset_id: T::IbcDenomToAssetIdConversion::from_denom_to_asset_id(
-					&packet_data.token.denom.to_string(),
-				)
-				.ok(),
-				amount: packet_data.token.amount.as_u256().as_u128().into(),
-				is_sender_source: is_sender_chain_source(
-					packet.source_port.clone(),
-					packet.source_channel.clone(),
-					&packet_data.token.denom,
-				),
-				source_channel: packet.source_channel.to_string().as_bytes().to_vec(),
-				destination_channel: packet.destination_channel.to_string().as_bytes().to_vec(),
-			}),
+			Ok(_) => {
+				let sequence: u64 = packet.sequence.into();
+				if <Pallet<T> as Store>::SequenceFee::contains_key(sequence) {
+					<Pallet<T> as Store>::SequenceFee::remove(sequence);
+				}
+				Pallet::<T>::deposit_event(Event::<T>::TokenTransferCompleted {
+					from: packet_data.sender,
+					to: packet_data.receiver,
+					ibc_denom: packet_data.token.denom.to_string().as_bytes().to_vec(),
+					local_asset_id: T::IbcDenomToAssetIdConversion::from_denom_to_asset_id(
+						&packet_data.token.denom.to_string(),
+					)
+					.ok(),
+					amount: packet_data.token.amount.as_u256().as_u128().into(),
+					is_sender_source: is_sender_chain_source(
+						packet.source_port.clone(),
+						packet.source_channel.clone(),
+						&packet_data.token.denom,
+					),
+					source_channel: packet.source_channel.to_string().as_bytes().to_vec(),
+					destination_channel: packet.destination_channel.to_string().as_bytes().to_vec(),
+				})
+			},
 			Err(e) => {
+				use ibc::{applications::transfer::context::BankKeeper, bigint::U256};
+				use sp_core::Get;
+
 				log::trace!(
 					target: "pallet_ibc::transfer",
 					"error: acknowledgement error: {e}",
 				);
+				let sequence: u64 = packet.sequence.into();
+				if <Pallet<T> as Store>::SequenceFee::contains_key(sequence) {
```

Comment:
can we change to SequenceFee::<T>::contains_key...

Reply:
hm. do not sure what you mean.
i tried but this syntax did not work.

Reply:
i meant SequenceFee::<T>::contains_key...

Reply:
Yes. Thanks. it works.
`crate::SequenceFee::<T>::contains_key`
at the first look it is not explicit to which pallet this storage belongs. is it storage or some random struct with contains_key.
for example when we are emitting event we are specifying the the Pallet
`Pallet::<T>::deposit_event`

Reply:
Yes. You are right! 
it is better to use the same approach across all the codebase. 
Fixed.  Done